### PR TITLE
Unescape paths for GlobPath

### LINKED ISSUE #138: Can't find any entries when the path contains spaces
I'm having problems with CtrlP since upgrading to Vim 7.4. 

CtrlP can't find any entries in some folders, this problem didn't occur before upgrading.

I think that the problem is that the path have spaces in it. Though this weren't a problem before upgrading from the built-in Vim in OS X (El Capitan). If I rename the folders in the path so that there are no spaces, then I don't have the problem anymore. But this shouldn't be required.

My `vim.rc`:

```
" -------- Vundle -------
set nocompatible              " be iMproved, required
filetype off                  " required

" set the runtime path to include Vundle and initialize
set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()
" alternatively, pass a path where Vundle should install plugins
"call vundle#begin('~/some/path/here')

" let Vundle manage Vundle, required
Plugin 'VundleVim/Vundle.vim'
Plugin 'kien/ctrlp.vim'

" All of your Plugins must be added before the following line
call vundle#end()            " required
filetype plugin indent on    " required
```

I upgraded using `brew install vim` and this is what `vim --version` says:

```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Oct  1 2015 17:36:26)
MacOS X (unix) version
Included patches: 1-884
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl             +farsi           +mouse_netterm   +syntax
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
-balloon_eval    +float           +mouse_urxvt     -tag_any_white
-browse          +folding         +mouse_xterm     -tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         -gettext         -mzscheme        +textobjects
-clientserver    -hangul_input    +netbeans_intg   +title
+clipboard       +iconv           +path_extra      -toolbar
+cmdline_compl   +insert_expand   +perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python          +viminfo
+cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con      -lua             +rightleft       +windows
+diff            +menu            +ruby            +writebackup
+digraphs        +mksession       +scrollbind      -X11
-dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     -xim
+emacs_tags      -mouseshape      -sniff           -xsmp
+eval            +mouse_dec       +startuptime     -xterm_clipboard
+ex_extra        -mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    -xpm
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: /usr/bin/clang -c -I. -Iproto -DHAVE_CONFIG_H   -F/usr/local/Frameworks -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -DMACOS_X_UNIX  -Os -w -pipe -march=native -mmacosx-version-min=10.11 -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: /usr/bin/clang   -L. -L/usr/local/lib -L/usr/local/lib -F/usr/local/Frameworks -Wl,-headerpad_max_install_names -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -F/private/tmp/vim20151001-44506-3ot6hp/vim-7.4.884/Frameworks -o vim        -lm  -lncurses -liconv -framework Cocoa   -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl -framework Python   -lruby.2.0.0 -lobjc
```

### PR DESCRIPTION
**Pull request by mattn**
Closes #138, #572

### CODE REVIEW

File: autoload/ctrlp.vim [hunk 1]
```diff
@@ -428,7 +428,8 @@ endf
 if has('patch-8.2-0995')
 	fu! s:GlobPath(dirs, depth)
 		let entries = []
-		for e in split(a:dirs, ',')
+		let dirs = substitute(a:dirs, '\\\([%#]\)', '\1', 'g')
```
Comment (tacahiroy) [inline autoload/ctrlp.vim:431]: This doesn't fix #138.
The second argument should be `'\\\([%# ]\)'`. (white-space is missing here)
A patch you commented on a ticket #138 is correct, but this MR doesn't contain that.

Comment (mattn) [reply autoload/ctrlp.vim:431]: Right. I'll fix.

**Review by tacahiroy — COMMENTED** [review]
It looks some more characters are possibly escaped in [ctrlp#utils#fnesc](https://github.com/ctrlpvim/ctrlp.vim/blob/61e964f2d168f52cb8e80e67d26605b52d726b16/autoload/ctrlp/utils.vim) and causes the same error.
I think characters which are escaped with `fnameescape` should be unescaped as well to eliminate possible errors.
What do you think?

**Comment by tacahiroy** [comment]
Confirmed your latest patch fixes issues #138 and #572.

**Comment by mattn** [comment]
>What do you think?

escape part is complex that is given parametes for way to specify escape-pattern. But unescap part does not know the escaped-pattern. So this is harder to work perfectly. However this patch is  less wrong. I think.

**Comment by tacahiroy** [comment]
> > What do you think?
> 
> escape part is complex that is given parametes for way to specify escape-pattern. But unescap part does not know the escaped-pattern. So this is harder to work perfectly. However this patch is less wrong. I think.

Indeed. And I think this fixes very popular problematic cases, so this is enough for now.
Thanks for fixing issues which had been listed for a long time.

**Pull request merged**